core/src/main/java/org/springframework/security/authentication/DefaultAuthenticationEventPublisher.java

```diff
@@ -160,6 +160,25 @@ public void setAdditionalExceptionMappings(Properties additionalExceptionMapping
 		}
 	}
 
+	/**
+	 * Overload of {@link #setAdditionalExceptionMappings(Properties)}}, providing the ability
+	 * to map authentication exceptions and failure authentication events using a {@link Map}
+	 *
+	 * @param mappings where keys are exception classes and values are event classes.
+	 */
+	public void setAdditionalExceptionMappings(Map<Class<? extends AuthenticationException>,
+			Class<? extends AbstractAuthenticationFailureEvent>> mappings){
+		Assert.notEmpty(mappings, "The mappings Map must not be empty nor null");
+		for (Map.Entry<Class<? extends AuthenticationException>,Class<? extends AbstractAuthenticationFailureEvent>> entry
+				: mappings.entrySet()) {
+				Class<?> exceptionClazz = entry.getKey();
```

Comment:
For consistency, would you please change the variable names here so they are the same as the other methods in this class? I believe they refer to the exception class as `exceptionClass`.